lib/oplog/tail.go

```diff
@@ -372,11 +366,42 @@ func closeCursor(cursor *mongo.Cursor) {
 // ignored it or failed at some later step.
 func (tailer *Tailer) unmarshalEntry(rawData bson.Raw, denylist *sync.Map, readOrdinal int) (timestamp *primitive.Timestamp, pubs []*redispub.Publication, sendMetricsData func()) {
 	var result rawOplogEntry
+	var ok bool
+	result.Namespace, ok = rawData.Lookup("ns").StringValueOK()
+	if !ok {
+		log.Log.Error("Error unmarshalling oplog namespace entry")
+	}
 
-	err := bson.Unmarshal(rawData, &result)
-	if err != nil {
-		log.Log.Errorw("Error unmarshalling oplog entry", "error", err)
-		return
```

Comment:
> Did it error if the struct field type didn't match the type in the document?
Yes, I tested this.  If a field was missing or extra, there was no error.  Only if a field existed but with a different type would the original unmarshal error.   So this tries to mostly follow that pattern, but does error on namespace which would seem to be a required field.  However, maybe none are -- I don't know what namespace would be set to in the case of a transaction.  In that case, splitting up the lookup and type conversion and exactly matching the old behavior is probably best -- I had started that way, but wanted to avoid it since it makes for terribly messy code.

Reply:
That would work -- I'd also have to pass in the denylist to allow for the early filtering, but that's probably fine